Sources/StreamVideo/WebRTC/v2/PeerConnection/MediaAdapters/ScreenShareMediaAdapter.swift

```diff
@@ -0,0 +1,170 @@
+        peerConnectionFactory: PeerConnectionFactory,
+        sfuAdapter: SFUAdapter,
+        videoOptions: VideoOptions,
+        videoConfig: VideoConfig,
+        subject: PassthroughSubject<TrackEvent, Never>,
+        screenShareSessionProvider: ScreenShareSessionProvider
+    ) {
+        self.init(
+            sessionID: sessionID,
+            peerConnection: peerConnection,
+            peerConnectionFactory: peerConnectionFactory,
+            localMediaManager: LocalScreenShareMediaAdapter(
+                sessionID: sessionID,
+                peerConnection: peerConnection,
+                peerConnectionFactory: peerConnectionFactory,
+                sfuAdapter: sfuAdapter,
+                videoOptions: videoOptions,
+                videoConfig: videoConfig,
+                subject: subject,
+                screenShareSessionProvider: screenShareSessionProvider
+            ),
+            subject: subject
+        )
+    }
+
+    init(
+        sessionID: String,
+        peerConnection: RTCPeerConnection,
+        peerConnectionFactory: PeerConnectionFactory,
+        localMediaManager: LocalMediaAdapting,
+        subject: PassthroughSubject<TrackEvent, Never>
+    ) {
+        self.sessionID = sessionID
+        self.peerConnection = peerConnection
+        self.peerConnectionFactory = peerConnectionFactory
+        self.localMediaManager = localMediaManager
+        self.subject = subject
+
+        peerConnection
+            .publisher(eventType: RTCPeerConnection.AddedStreamEvent.self)
+            .filter { $0.stream.trackType == .screenshare }
+            .sink { [weak self] in self?.add($0.stream) }
+            .store(in: disposableBag)
+
+        peerConnection
+            .publisher(eventType: RTCPeerConnection.RemovedStreamEvent.self)
+            .filter { $0.stream.trackType == .screenshare }
+            .sink { [weak self] in self?.remove($0.stream) }
+            .store(in: disposableBag)
+    }
+
+    // MARK: - MediaAdapting
+
+    func setUp(
+        with settings: CallSettings,
+        ownCapabilities: [OwnCapability]
+    ) async throws {
+        try await localMediaManager.setUp(
+            with: settings,
+            ownCapabilities: ownCapabilities
+        )
+    }
+
+    func didUpdateCallSettings(
+        _ settings: CallSettings
+    ) async throws {
+        try await localMediaManager.didUpdateCallSettings(settings)
+    }
+
+    // MARK: - ScreenSharing
+
+    func beginScreenSharing(
+        of type: ScreensharingType,
+        ownCapabilities: [OwnCapability]
+    ) async throws {
+        guard
+            let localScreenShareMediaManager = localMediaManager as? LocalScreenShareMediaAdapter
+        else {
+            return
+        }
+
+        try await localScreenShareMediaManager.beginScreenSharing(
+            of: type,
+            ownCapabilities: ownCapabilities
+        ) { [weak self] in
+//            self?.queue.sync { [weak self] in
```

Comment:
no op here?

Reply:
That's not needed anymore and thus has been removed (local streams are managed by the transceiver) and remote tracks lifecycle is managed by the SFU events.